Update doc on profiling config settings (#49964)
dotnet/docs: docs/core/runtime-config/debugging-profiling.md

- Configures whether profiling is enabled for the currently running process.
- If you omit this setting, profiling is disabled. This is equivalent to setting the value to `0`.

> - In addition to enabling profiling, the profiler GUID and profiler location also need to be configured using the settings below in order to load a profiler.

Reviewer: ```suggestion
- To load a profiler, in addition to enabling profiling, the profiler GUID and profiler location also need to be configured using these settings.
```

| | Setting name | Values |
| - | - | - |
| **runtimeconfig.json** | N/A | N/A |
| **Environment variable** | `CORECLR_PROFILER` or `DOTNET_PROFILER` | *string-guid* |

## Profiler location

Once profiling is enabled, the profiler can be loaded in two ways, with environment variables (cross-plat) or through the registry (Windows only).

### Environment Variable (cross-plat)

- Specifies the path to the profiler DLL to load into the currently running process (or 32-bit or 64-bit process).
- If more than one variable is set, the bitness-specific variables take precedence. They specify which bitness of profiler to load.

> - If any of these environment variables are present, the registry look up is skipped altogether, and the path from `DOTNET_PROFILER_PATH` is used to load the DLL.

Reviewer 2: ```suggestion
```
It might be any of the environment variables that take precedence, not solely DOTNET_PROFILER_PATH. I suggested a different way we could phrase it just above the environment variable section.

> - When `DOTNET_PROFILER_PATH` is specified _and_ the profiler is registered, `DOTNET_PROFILER_PATH` always takes precedence. Even if `DOTNET_PROFILER_PATH` points to an invalid path, the runtime still uses `DOTNET_PROFILER_PATH`, and fails to load the profiler.

Reviewer 2: ```suggestion
```
This seems like it is duplicating the message that environment variables take precedence over registry entries.

> - `DOTNET_PROFILER` is _always required_. When `DOTNET_PROFILER_PATH` is specified, the registry look up is skipped. The runtime still needs the profiler's CLSID (GUID) to pass it to the class factory's CreateInstance call.

Author: Done.

## Profiler location

Once profiling is enabled, the profiler can be loaded in two ways, with environment variables (cross-plat) or through the registry (Windows only). The profiler path environment variables take precedence over any COM library path in the registry if both are specified.

### Environment variable (cross-plat)

- Specifies the path to the profiler DLL to load into the currently running process (or 32-bit or 64-bit process).
- If more than one variable is set, the bitness-specific variables take precedence. They specify which bitness of profiler to load.

| | Setting name | Values |
| - | - | - |
| **Environment variable** | `CORECLR_PROFILER_PATH` or `DOTNET_PROFILER_PATH` | *string-path* |
| **Environment variable** | `CORECLR_PROFILER_PATH_32` or `DOTNET_PROFILER_PATH_32` | *string-path* |
| **Environment variable** | `CORECLR_PROFILER_PATH_64` or `DOTNET_PROFILER_PATH_64` | *string-path* |
| **Environment variable** | `CORECLR_PROFILER_PATH_ARM32` or `DOTNET_PROFILER_PATH_ARM32` | *string-path* |
| **Environment variable** | `CORECLR_PROFILER_PATH_ARM64` or `DOTNET_PROFILER_PATH_ARM64` | *string-path* |

### Through the registry (Windows only)

> When the `DOTNET_PROFILER_PATH*` environment variables above are not set while running on Windows, coreclr looks up the CLSID from `DOTNET_PROFILER` in the registry to find the full path to the profiler's DLL. Just like with any COM server DLL, the profiler's CLSID is looked up under HKEY_CLASSES_ROOT, which merges the classes from HKLM and HKCU.

Reviewer: ```suggestion
When the `DOTNET_PROFILER_PATH*` [environment variables](#environment-variable--cross-plat-) aren't set while running on Windows, coreclr looks up the CLSID from `DOTNET_PROFILER` in the registry to find the full path to the profiler's DLL. Just like with any COM server DLL, the profiler's CLSID is looked up under HKEY_CLASSES_ROOT, which merges the classes from HKLM and HKCU.
```
